odinochka.js

```diff
@@ -15,22 +15,57 @@ function newTabs(data) {
     data.tabs.forEach(o => chrome.tabs.create({url: o.url, pinned: o.pinned}))
 }
 
-function cssfilter(x) {
-    let node = document.getElementById("cssfilterstyle");
-    if(x.target.value != "") {
-        node.innerHTML = `a.tab:not([href*="${x.target.value}"]) {display:none} `;
-        // TODO someday when :has works, also hide the empty groups
-    }
-    else {
-        node.innerHTML = "";
+function debounce(func, wait, immediate) {
+    let timeout;
+    return function(e) {
+        const context = e,
+            args = arguments;
+        const later = () => {
+            timeout = null;
+            if (!immediate) func.apply(context, args);
+        };
+        const callNow = immediate && !timeout;
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+        if (callNow) func.apply(context, args);
+    };
+}
+
+function cssfilter(e) {
+    // render entire list from indexDB if input is empty
+    var query = e.target.value
+    if (query === '') {
+        render()
+        return
     }
+    window.indexedDB.open("odinochka", 5).onsuccess = function(event){
+        let db = event.target.result;
+
+        let tx = db.transaction('tabgroups', 'readonly');
+        let store = tx.objectStore('tabgroups');
+
+        let allRecords = store.getAll()
+        allRecords.onsuccess = function() {
+            var filteredAppData = allRecords.result.filter(function(cursor) {
```

Comment:
went a little crazy with the `var`, `let`, `const` syntax here - feel free to change these all to `var` for consistency